### ISSUE TITLE
Building with E57 enabled causes: fatal error: 'E57Foundation.h' file not found

### ISSUE THREAD

**Issue by ryanfb** [CloudCompare/CloudCompare#711]
## Current Behavior

If you enable E57 support in CMake with the default included submodule path for libE57Format, running `make` halts with the error:

```
[ 30%] Building CXX object libs/qCC_io/CMakeFiles/QCC_IO_LIB.dir/E57Filter.cpp.o
In file included from /Users/ryan/source/cloudcompare/libs/qCC_io/E57Filter.cpp:23:
/Users/ryan/source/cloudcompare/libs/qCC_io/E57Header.h:4:10: fatal error: 'E57Foundation.h' file not found
#include <E57Foundation.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [libs/qCC_io/CMakeFiles/QCC_IO_LIB.dir/E57Filter.cpp.o] Error 1
make[1]: *** [libs/qCC_io/CMakeFiles/QCC_IO_LIB.dir/all] Error 2
make: *** [all] Error 2
```

As a workaround, you can edit `E57Header.h` and `E57Filter.cpp` to use an absolute path e.g.:

    #include "/Users/ryan/source/cloudcompare/contrib/libE57Format/include/E57Foundation.h"

Instead of `#include <E57Foundation.h>`.

## Expected Behavior

The CMake configuration process should put `E57Foundation.h` on the include path correctly if E57 support is enabled and the default libE57Format path is set.

## Steps to Reproduce (for bugs)

* `git clone --recursive https://github.com/cloudcompare/trunk.git cloudcompare`
* `cd cloudcompare`
* `mkdir build`
* `cd build`
* `ccmake ..` (press c, c, g, etc.)
* `make`

## Context/Comments

Maybe this is related to using an out-of-place build (i.e. `build` directory), and the library is on the path for an in-place build?

## Your Environment
* CC Version: HEAD / 07677788e2fb1cadeb8a48e15bab6b206b9d4ab6
* OS & Version: OS X 10.12.6

**Comment by asmaloney** [comment]
I think if you change the include the way you are you'll still end up with a link error?

The ilbE57Format lib must be built and installed already, then you set `LIBE57FORMAT_INSTALL_DIR` to point at the directory with the include/ and lib/ in it.

I will try to make cmake smarter about this so it catches incorrect configuration at cmake time rather than compile/link time.

**Comment by asmaloney** [comment]
Aside: It looks like it's possible to build the dependent submodule from the main cmake files using [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) but I'm afraid that's beyond my cmake skills at the moment.

**Comment by ryanfb** [comment]
I don't have any problems linking (or using the resulting CloudCompare.app to open E57 files) when I use an absolute path for `E57Foundation.h`. For the include file, it doesn't/shouldn't seem to matter if the libE57Format lib is already built or not, since `contrib/libE57Format/include/E57Foundation.h` is there regardless of if it's been built. Though not having it built will later halt the `make` process during link time with `make[2]: *** No rule to make target `../contrib/libE57Format/lib/libE57Format.a', needed by `libs/qCC_io/libQCC_IO_LIB.dylib'.  Stop.`, until you build it so that `lib/libE57Format.a` exists.

**Comment by asmaloney** [comment]
If it's built and installed properly, and LIBE57FORMAT_INSTALL_DIR is set properly, it will find the header file without changing the code:

```
if( NOT LIBE57FORMAT_INSTALL_DIR )
	message( SEND_ERROR "No libE57Format install dir specified (LIBE57FORMAT_INSTALL_DIR)" )
else()
	include_directories( ${LIBE57FORMAT_INSTALL_DIR}/include/E57Format )
endif()
```

What is your `CMAKE_INSTALL_PREFIX` for libeE57Format set to? That's what you need to put in CloudComapre's `LIBE57FORMAT_INSTALL_DIR`.

**Comment by ryanfb** [comment]
Ok, I see what's going on now. I was just using `libE57Format` in-place in the `contrib` directory since that's what `LIBE57FORMAT_INSTALL_DIR` defaults to, by building the CMake project for it in `lib` and running `make` with no `make install`. Since the `make install` target also puts `E57Foundation.h` under `include/E57Format/E57Foundation.h` instead of just `include`, it wasn't detected by that CloudCompare CMake `include_directories` rule which expects it in `E57Format`.

So you could get around this by setting `CMAKE_INSTALL_PREFIX` for libE57Format to the in-place contribu/submodule directory before using `cmake` to build libE57Format and running `make install` for libE57Format before building CloudCompare.

**Comment by asmaloney** [comment]
Yes - but you can set `CMAKE_INSTALL_PREFIX` to whatever you want. I don't build anything inside the CloudCompare directory. I actually build all my support libs separately and install them in their own directory that I then point CloudCompare at.

I'm changing the cmake on the CloudCompare side to not include a default directory so it's clear(er) that you need to set it. I'm also going to put in some smarts in about making sure the libs exist.

Ideally we should use the `ExternalProject` stuff I mentioned above...
